Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeLists.txt

```diff
@@ -872,10 +884,36 @@ add_custom_command (
     COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:H5detect>
     ARGS > ${HDF5_BINARY_DIR}/H5Tinit.c
     DEPENDS ${INTERNAL_BUILD_PREFIX}H5detect
-)
+    WORKING_DIRECTORY ${HDF5_GENERATED_SOURCE_DIR}
+  )
+  if (BUILD_SHARED_LIBS)
+    if (NOT EXISTS ${HDF5_GENERATED_SOURCE_DIR}/shared/H5Tinit.c)
+      add_custom_command (
+          OUTPUT ${HDF5_GENERATED_SOURCE_DIR}/shared/H5Tinit.c
```

Comment:
This `CMakeLists.txt` file already has a similar rule:

```cmake
  add_custom_command (
      OUTPUT ${HDF5_BINARY_DIR}/shared/H5Tinit.c
```

This is the cause of the CMake Error about duplicate custom rules.

Reply:
The error does not occur if HDF5 is built on its own. The problem appears when it is bundled in ITK.

Reply:
I suspect that ITK's modifications cause the duplicate code, perhaps due to incorrect conflict resolution.